Helper/GraphQL/Client.php

```diff
@@ -0,0 +1,197 @@
+     *
+     * @var RequestFactory
+     */
+    private $requestFactory;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @param Context $context
+     * @param ZendClientFactory $httpClientFactory
+     * @param ConfigHelper $configHelper
+     * @param ResponseFactory $responseFactory
+     * @param RequestFactory $requestFactory
+     * @param LogHelper $logHelper
+     * @param Bugsnag $bugsnag
+     * @codeCoverageIgnore
+     */
+    public function __construct(
+        Context $context,
+        ZendClientFactory $httpClientFactory,
+        ConfigHelper $configHelper,
+        ResponseFactory $responseFactory,
+        RequestFactory $requestFactory,
+        LogHelper $logHelper,
+        Bugsnag $bugsnag
+    ) {
+        parent::__construct($context);
+        $this->httpClientFactory = $httpClientFactory;
+        $this->configHelper = $configHelper;
+        $this->responseFactory = $responseFactory;
+        $this->requestFactory = $requestFactory;
+        $this->logHelper = $logHelper;
+        $this->bugsnag = $bugsnag;
+    }
+
+
+    private function makeGQLCall($query, $operation, $variables) {
+        $result = $this->responseFactory->create();
+        $client = $this->httpClientFactory->create();
+
+        $apiKey = $this->configHelper->getApiKey();
+
+        $gqlRequest = array(
+            "operationName" => $operation,
+            "variables" => $variables,
+            "query" => $query
+        );
+
+        $requestData = json_encode($gqlRequest, JSON_UNESCAPED_SLASHES);
+
+        $apiURL = $this->configHelper->getApiUrl() . self::MERCHANT_API_GQL_ENDPOINT;
+
+        $client->setUri($apiURL);
+
+        $client->setConfig(['maxredirects' => 0, 'timeout' => 30]);
+
+        $headers =  ApiUtils::constructRequestHeaders(
+            $this->configHelper->getStoreVersion(),
+            $this->configHelper->getModuleVersion(),
+            $requestData,
+            $apiKey,
+            array()
+        );
+
+        $client->setHeaders($headers);
+
+        $responseBody = null;
+
+        try {
+            $response     = $client->setRawData($requestData, 'application/json')->request("POST");
+            $responseBody = $response->getBody();
+
+            $this->bugsnag->registerCallback(function ($report) use ($response) {
+                $report->setMetaData([
+                    'BOLT API RESPONSE' => [
+                        'headers' => $response->getHeaders(),
+                        'body'    => $response->getBody(),
+                    ]
+                ]);
+            });
+
+            $this->bugsnag->registerCallback(function ($report) use ($response) {
+                $headers = $response->getHeaders();
+                $report->setMetaData([
+                    'META DATA' => [
+                        'bolt_trace_id' => @$headers[ConfigHelper::BOLT_TRACE_ID_HEADER],
+                    ]
+                ]);
+            });
```

Comment:
You can probably join these two callbacks into one.

Reply:
one uses $request adn the the other $response? so maybe not. re-open if you feel strongly. Also i want to do one more refactoring pass but will do it once i get this in.